test/sparse/sparsevector.jl

```diff
@@ -177,13 +177,33 @@ end
 # generic array index
 let x = sprand(100, 0.5)
     I = rand(1:length(x), 20)
-    @which x[I]
     r = x[I]
     @test isa(r, SparseVector{Float64,Int})
     @test all(nonzeros(r) .!= 0.0)
     @test Array(r) == Array(x)[I]
 end
 
+# boolean array index
+let x = sprand(10, 10, 0.5)
+    I = rand(1:size(x, 2), 10)
+    bI = falses(size(x, 2))
+    bI[I] = true
+    r = x[1,bI]
+    @test isa(r, SparseVector{Float64,Int})
+    @test all(nonzeros(r) .!= 0.0)
```

Comment:
`@kshyatt` resolved my confusion (thanks!): This test ascertains that [entries in the sliced sparse matrix that (1) the logical index set selects for retention in the result and (2) are not stored in the sliced sparse matrix] are not stored in the result. Best!